Compile error type mismatch?

```
#cargo build  --target arm-unknown-linux-gnueabi 
   Compiling users v0.5.3
   Compiling pam-sys v0.5.4
   Compiling pam-auth v0.5.3
error[E0308]: mismatched types
  --> /Users/xxx/.cargo/registry/src/github.com-1ecc6299db9ec823/pam-auth-0.5.3/src/env.rs:56:46
   |
56 |         unsafe { pam::raw::pam_misc_drop_env(self.ptr as *mut *mut i8) };
   |                                              ^^^^^^^^^^^^^^^^^^^^^^^^ expected u8, found i8
   |
   = note: expected type `*mut *mut u8`
              found type `*mut *mut i8`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0308`.
error: Could not compile `pam-auth`.

```

--- a/src/env.rs
+++ b/src/env.rs
@@ -51,8 +51,9 @@ impl PamEnvList {
     }
 }
 
+#[cfg(target_os = "linux")]
 impl Drop for PamEnvList {
     fn drop(&mut self) {
-        unsafe { pam::raw::pam_misc_drop_env(self.ptr as *mut *mut i8) };
+        unsafe { pam::raw::pam_misc_drop_env(self.ptr as *mut *mut c_char) };
     }
 }

